### ISSUE TITLE
你好，波哥，请问T5-PEGASUS的generate策略是和transformers库的mt5不同吗？

### ISSUE THREAD

**Issue by fuhaito7** [Tongjilibo/bert4torch#35]
我使用同样的权重文件，同样的测试数据，使用如下代码可以生成摘要：
model = MT5ForConditionalGeneration.from_pretrained(args.pretrain_model).to(device)
gen = model.generate(max_length=args.max_len_generate,
                                eos_token_id=tokenizer.sep_token_id,
                                decoder_start_token_id=tokenizer.cls_token_id,
                                **content)

使用Autotitle  示例中的generate输出为空，后来发现是beam_search中预测的第一个值就是end_id [SEP]
class AutoSummarize(AutoRegressiveDecoder):
    """seq2seq解码器
    """
    @AutoRegressiveDecoder.wraps(default_rtype='logits')
    def predict(self, inputs, output_ids, states):
        # inputs中包含了[decoder_ids, encoder_hidden_state, encoder_attention_mask]
        return model.decoder.predict([output_ids] + inputs)[-1][:, -1, :]  # 保留最后一位

    def generate(self, text):
        gc.collect()
        torch.cuda.empty_cache()
        token_ids, _ = tokenizer.encode(text, maxlen=args.max_c_len)
        token_ids = torch.tensor([token_ids], device=device)
        encoder_output = model.encoder.predict([token_ids])
        output_ids = self.beam_search(encoder_output, topk=1)  # 基于beam search
        return tokenizer.decode([int(i) for i in output_ids.cpu().numpy()]

不知道是不是因为states为空呢，我发现encoder_output里并没有states

**Comment by Tongjilibo** [comment]
可以git clone下最新代码，然后跑一下这个示例看看
[t5预训练模型测试](https://github.com/Tongjilibo/bert4torch/blob/master/examples/basic/basic_language_model_t5_pegasus.py)
[权重转换](https://github.com/Tongjilibo/bert4torch/blob/master/examples/convert_script/convert_t5_pegasus.py)

**Comment by fuhaito7** [comment]
![image](https://user-images.githubusercontent.com/91183532/191179831-74d91bb0-74ca-497a-b5f1-e5ed26a50805.png)
还是一样的

**Comment by Tongjilibo** [comment]
[权重转换](https://github.com/Tongjilibo/bert4torch/blob/master/examples/convert_script/convert_t5_pegasus.py)
里面的config也要换一下，你再试下
```
# base版本
{
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 768,
  "initializer_range": 0.02,
  "intermediate_size": 2048,
  "num_attention_heads": 12,
  "attention_head_size": 64,
  "num_hidden_layers": 12,
  "vocab_size": 50000,
  "relative_attention_num_buckets": 32,
  "attention_scale":  false,
  "is_dropout": true
}
# small版本
{
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 512,
  "initializer_range": 0.02,
  "intermediate_size": 1024,
  "num_attention_heads": 6,
  "attention_head_size": 64,
  "num_hidden_layers": 8,
  "vocab_size": 50000,
  "relative_attention_num_buckets": 32,
  "attention_scale":  false,
  "is_dropout": true
}
```

**Comment by fuhaito7** [comment]
这次可以了，非常感谢大佬，
